mnoptical/examples/linear_topo_watts_strogatz.py

```diff
@@ -0,0 +1,477 @@
+    else:
+        with open(channel_file, newline='') as f:
+            channels = [list(map(int, rec)) for rec in csv.reader(f, delimiter=',')]
+    print(channels)
+    conn1_file = open(f"first__connection_topo2_p{prob_file}.csv", 'w')
+    conn1_writer = csv.writer(conn1_file)
+    conn2_file = open(f"second__connection_topo2_p{prob_file}.csv", 'w')
+    conn2_writer = csv.writer(conn2_file)
+    conn3_file = open(f"third__connection_topo2_p{prob_file}.csv", 'w')
+    conn3_writer = csv.writer(conn3_file)
+    conn4_file = open(f"fourth__connection_topo2_p{prob_file}.csv", 'w')
+    conn4_writer = csv.writer(conn4_file)
+    conn5_file = open(f"fifth__connection_topo2_p{prob_file}.csv", 'w')
+    conn5_writer = csv.writer(conn5_file)
+    conn6_file = open(f"sixth__connection_topo2_p{prob_file}.csv", 'w')
+    conn6_writer = csv.writer(conn6_file)
+    conn7_file = open(f"seventh__connection_topo2_p{prob_file}.csv", 'w')
+    conn7_writer = csv.writer(conn7_file)
+    conn8_file = open(f"eighth__connection_topo2_p{prob_file}.csv", 'w')
+    conn8_writer = csv.writer(conn8_file)
+    create_header(conn1_writer, channels_length)
+    create_header(conn2_writer, channels_length)
+    create_header(conn3_writer, channels_length)
+    create_header(conn4_writer, channels_length)
+    create_header(conn5_writer, channels_length)
+    create_header(conn6_writer, channels_length)
+    create_header(conn7_writer, channels_length)
+    create_header(conn8_writer, channels_length)
+    conn = [{'start': 1, 'end': 13, 'ch': [], 'file': conn1_writer}, {'start': 2, 'end': 17, 'ch': [], 'file': conn2_writer},
+            {'start': 4, 'end': 16, 'ch': [], 'file': conn3_writer}, {'start': 5, 'end': 18, 'ch': [], 'file': conn4_writer},
+            {'start': 3, 'end': 15, 'ch': [], 'file': conn5_writer}, {'start': 6, 'end': 19, 'ch': [], 'file': conn6_writer},
+            {'start': 7, 'end': 20, 'ch': [], 'file': conn7_writer}, {'start': 8, 'end': 14, 'ch': [], 'file': conn8_writer}]
+    topo = LinearTopo(N=20, p=prob, connection=conn)
+    net = Mininet(topo=topo)
+    print("starting model --- ")
+    restServer = RestServer(net)
+    net.start()
+    restServer.start()
+    requestHandler = RESTProxy()
+    plotNet(net, outfile=f'eight_conn_linear_topo2-watts_plot_{prob_file}.png', directed=True)
+    counter = 1
+    for ch in range(channels_length):
+        count = 0
+        for end_conn in conn:
+            configNet(net, connection_detail[count], end_conn['start'], end_conn['end'], counter, [channels[count][ch]])
+            end_conn['ch'].append(channels[count][ch])
+            count += 1
+        start_transceiver(net, conn)
+        monitorOSNR(requestHandler, conn, connection_detail)
+        counter += 1
+    conn1_file.close()
+    conn2_file.close()
+    conn3_file.close()
+    conn4_file.close()
+    conn5_file.close()
+    conn6_file.close()
+    conn7_file.close()
+    conn8_file.close()
+    if 'test' in argv:
```

Comment:
@juraul1 I have updated the requirements.txt files with the deps, but the test is failing because of incomplete test cases. Do I have to write the test case for this setup, or can I comment out this test snippet?

Reply:
Oh yes you can remove that!